src/webapi/rendering_context.rs

```diff
@@ -27,7 +183,771 @@ impl RenderingContext for CanvasRenderingContext2d {
     }
 }
 
+impl CanvasGradient {
+
+    /// Adds a new stop, defined by an offset and a color, to the gradient. If the offset is 
+    /// not between 0 and 1, an INDEX_SIZE_ERR is raised, if the color can't be parsed as a 
+    /// CSS <color>, a SYNTAX_ERR is raised.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient/addColorStop)
+    // https://html.spec.whatwg.org/#2dcontext:dom-canvasgradient-addcolorstop
+    pub fn add_color_stop(&self, offset: f64, color: &str) -> Result<(), SyntaxError> {
+        assert!(offset > 0 as f64 && offset < 1 as f64);
+        js_try! ( @(no_return)
+            @{&self.0}.addColorStop(@{offset}, @{color});
+        ).unwrap()
+    }
+}
+
+impl ImageData {
+
+    /// Returns a Uint8ClampedArray representing a one-dimensional array containing the data in the RGBA order, 
+    /// with integer values between 0 and 255 (included).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/data)
+    pub fn get_data(&self) -> Vec<u8> {
+        js! (
+            return @{&self.0}.data;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the number of rows in the image data object.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/height)
+    pub fn get_height(&self) -> f64 {
```

Comment:
1. These are missing links to the specs, e.g. for this method specifically `https://html.spec.whatwg.org/#2dcontext:dom-imagedata-width`.

2. In the specs this has type of `unsigned long`, so it should return `u32` instead of `f64`. Ditto for other methods. (This is why a link to the specs it useful - you can quickly check whenever the types are correct.)